src/util/references.ts

```
@@ -45,10 +43,13 @@ export function addNodeReference(
 ): void {
   let references = snapshot[direction];
   if (!references) {
-    references = (snapshot as any)[direction] = [];
+    references = snapshot[direction] = [];
   }
 
-  references.push({ id, path });
+  const idx = getIndexOfGivenReference(references, id, path);
+  if (idx === -1) {
```

I think I'd rather we throw an error in this case; AFAIK all places that call `addNodeReference` already have to guard against this case, or at least thought they were

---

This may not be true as now we have alias so it is possible that we have already add an alias (that gets de-alias) then encounter same field with different alias or in its true form.  I do think we shouldn't call the check before call this function (so we don't have to do unnecessary loop over array)